src/bazel/bazel_exit_code.ts

```diff
@@ -0,0 +1,137 @@
+// Copyright 2019 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// tslint:disable:max-line-length
+
+/**
+ * Bazel exit codes.
+ *
+ * This should be kept in sync with
+ *   https://source.bazel.build/bazel/+/master:src/main/cpp/util/exit_code.h
+ * and
+ *  https://source.bazel.build/bazel/+/master:src/main/java/com/google/devtools/build/lib/util/ExitCode.java
+ */
+export enum ExitCode {
+  SUCCESS = "SUCCESS",
+  BUILD_FAILURE = "BUILD_FAILURE",
+  PARSING_FAILURE = "PARSING_FAILURE",
+  COMMAND_LINE_ERROR = "COMMAND_LINE_ERROR",
+  TESTS_FAILED = "TESTS_FAILED",
+  PARTIAL_ANALYSIS_FAILURE = "PARTIAL_ANALYSIS_FAILURE",
+  NO_TESTS_FOUND = "NO_TESTS_FOUND",
+  RUN_FAILURE = "RUN_FAILURE",
+  ANALYSIS_FAILURE = "ANALYSIS_FAILURE",
+  INTERRUPTED = "INTERRUPTED",
+  LOCK_HELD_NOBLOCK_FOR_LOCK = "LOCK_HELD_NOBLOCK_FOR_LOCK",
+  REMOTE_ENVIRONMENTAL_ERROR = "REMOTE_ENVIRONMENTAL_ERROR",
+  OOM_ERROR = "OOM_ERROR",
+  REMOTE_ERROR = "REMOTE_ERROR",
+  LOCAL_ENVIRONMENTAL_ERROR = "LOCAL_ENVIRONMENTAL_ERROR",
+  INTERNAL_ERROR = "INTERNAL_ERROR",
+  PUBLISH_ERROR = "PUBLISH_ERROR",
+  REMOTE_EXECUTOR_OVERLOADED = "REMOTE_EXECUTOR_OVERLOADED",
+  RESERVED = "RESERVED",
+  UNKNOWN = "UNKNOWN",
+}
+
+export function exitCodeToUserString(exitCode: ExitCode): string {
```

Comment:
Please add JSDoc blocks for any function (exported or otherwise) that's not a trivial accessor/mutator.

Reply:
Done